src/Microsoft.VisualBasic/tests/VB/FileIOTests.vb

```diff
@@ -0,0 +1,897 @@
+﻿' Licensed to the .NET Foundation under one or more agreements.
+' The .NET Foundation licenses this file to you under the MIT license.
+' See the LICENSE file in the project root for more information.
+Option Explicit On
+Option Strict On
+
+Imports Microsoft.VisualBasic.FileIO
+Imports System.Collections.ObjectModel
+Imports System.Runtime.CompilerServices
+Imports System.Runtime.InteropServices
+Imports System.Text
+Imports Xunit
+' Do not Imports System.IO
+Namespace Microsoft.VisualBasic.Tests.VB
+    Public NotInheritable Class FileIOTests
+        Inherits FileIOTestBase
+
+        Sub New()
+        End Sub
+
+        Shared ReadOnly DestData() As Char = {"x"c, "X"c, "y"c}
+        Shared ReadOnly SourceData() As Char = {"a"c, "A"c, "b"c}
+
+        Public Shared ReadOnly Property ManualTestsEnabled() As Boolean
+            Get
+                Return Not String.IsNullOrEmpty(Environment.GetEnvironmentVariable("MANUAL_TESTS"))
+            End Get
+        End Property
+
+        Private Shared Function DirectoryListToString(DirectoryList As ReadOnlyCollection(Of String)) As String
+            Dim S As New StringBuilder
+            For Each d As String In DirectoryList
+                S.Append($"{d} ")
+            Next
+            Return S.ToString
+        End Function
+
+        Private Shared Function HasExpectedData(FileNameWithPath As String, ExpectedData() As Char) As Boolean
+            Using stream As New IO.StreamReader(IO.File.OpenRead(FileNameWithPath))
+                Dim ReadData(ExpectedData.Length - 1) As Char
+                stream.Read(ReadData, 0, SourceData.Length)
```

Comment:
`ExpectedData.Length`

Reply:
I am confused, VB uses 0 based indexes and does not have a trailing 0 like C# so I believe this is correct and the test passes.